### PR TITLE
[Testcase Refactoring] Enable PrivateUse1 tests and add hw-classification in test_transformer

### PR DESCRIPTION
**Pull request by PlatformLC**
## Summary
Enable the `test_ir` device-type variant for out-of-tree PrivateUse1 accelerators in `test/distributed/pipelining/test_transformer.py`, and classify `TransformerTests` as `HardwareClassification.ACCELERATOR`.

## Changes
- Drop the `only_for=["cpu", "cuda", "hpu", "xpu"]` argument (and the now-unused `devices` list) from `instantiate_device_type_tests`, keeping `allow_xpu=True`. With `only_for` removed, `get_desired_device_type_test_bases` auto-registers `PrivateUse1TestBase` when `_is_privateuse1_backend_available()`, so the PrivateUse1 variant of `test_ir` is generated alongside the existing cpu/cuda/hpu/xpu variants. Zero-loss: HPU is added unconditionally on `TEST_HPU`, XPU via `allow_xpu`, and the only addition is PrivateUse1.
- Set `hw_classification = HardwareClassification.ACCELERATOR` on `TransformerTests`. `test_ir` builds a model on `device`, runs `pipeline(...)` and asserts the pipelined output matches eager -- accelerator-generic behavior with no CUDA-specific paths. The classification is metadata: it only filters tests when `--hw-classification` is passed; normal discovery/execution is unchanged.

## Motivation
`test_ir` was never generated for out-of-tree PrivateUse1 accelerators because the explicit `only_for` list excluded PrivateUse1. Dropping it lets the framework's existing PrivateUse1 infrastructure (`PrivateUse1TestBase`, `_is_privateuse1_backend_available`) discover the backend automatically. The `HardwareClassification.ACCELERATOR` tag participates in the hardware-classification filtering initiative, so `--hw-classification ACCELERATOR` selects this test across every accelerator backend.

## Test Plan
```
python test/distributed/pipelining/test_transformer.py -v
python test/distributed/pipelining/test_transformer.py -v --hw-classification ACCELERATOR
python test/distributed/pipelining/test_transformer.py -v --hw-classification CPU
python test/distributed/pipelining/test_transformer.py -v --hw-classification GENERIC
```
Verified on a PrivateUse1 host (8 devices): the PrivateUse1 variant of `test_ir` is auto-generated (`TransformerTestsPRIVATEUSE1`) and passes alongside `test_ir_cpu` (normal run: Ran 2, OK). `--hw-classification ACCELERATOR` includes `test_ir` (total=2, passed=2); `--hw-classification CPU` and `--hw-classification GENERIC` filter it out (mismatched=2, Ran 0 each), confirming the filter is category-specific.

## Notes
This is part of the test case refactoring initiative tracked in #185590.

### CODE REVIEW

**Comment by PlatformLC** [comment]
@wjlFlyer @pjyFight @FuDdd @fffrog @lyriexs666 @jishuangfeng @JamesD18 @JiasenTian

File: test/distributed/pipelining/test_transformer.py [hunk 1]
```diff
@@ -74,10 +74,7 @@ def get_layers(module):
         print(f"Equivalence test passed {torch.sum(out)} ref {torch.sum(ref)}")
 
 
-devices = ["cpu", "cuda", "hpu", "xpu"]
-instantiate_device_type_tests(
-    TransformerTests, globals(), only_for=devices, allow_xpu=True
-)
+instantiate_device_type_tests(TransformerTests, globals(), allow_xpu=True)
```
Comment (drisspg) [inline test/distributed/pipelining/test_transformer.py:83]: this api is a lil weird to me alban like why is `allow_xpu` a kwarg  inclusion, whihle I assume cpu, cuda, hpu and xpu are all now default in this set of allowing? 

also our device test helpers have always confused me 

cc @albanD

**Comment by PlatformLC** [comment]
Same doubt on this, xpu/hpu/mps are using kwargs to enable while `only_for`is a confusing argument...
---- Replied Message ----
From Driss ***@***.***> Date 07/14/2026 00:43 To ***@***.***> Cc Chun ***@***.***>***@***.***> Subject Re: [pytorch/pytorch] [Test] Enable PrivateUse1 tests in test_transformer (PR #189682) 
@drisspg commented on this pull request.
In test/distributed/pipelining/test_transformer.py:
> @@ -74,10 +74,7 @@ def get_layers(module): print(f"Equivalence test passed {torch.sum(out)} ref {torch.sum(ref)}") -devices = ["cpu", "cuda", "hpu", "xpu"] -instantiate_device_type_tests( - TransformerTests, globals(), only_for=devices, allow_xpu=True -) +instantiate_device_type_tests(TransformerTests, globals(), allow_xpu=True) 
this api is a lil weird to me alban like why is allow_xpu a kwarg inclusion, whihle I assume cpu, cuda, hpu and xpu are all now default in this set of allowing?
also our device test helpers have always confused me
cc @albanD
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications, keep track of coding agent tasks and review pull requests on the go with GitHub Mobile for iOS and Android. Download it today!

File: test/distributed/pipelining/test_transformer.py (hunk 1 above)
Comment (albanD) [reply test/distributed/pipelining/test_transformer.py:83]: Historical reasons? haha
I think it's because xpu was added in core after the fact. Not much we can do about it now...

**Comment by albanD** [comment]
@PlatformLC as a new user I would recommend focussing on getting one PR merge before sending more.
For now, I'll close the others, feel free to re-open them once this one is good and you have more experience here.

File: test/distributed/pipelining/test_transformer.py (hunk 1 above)
Comment (PlatformLC) [reply test/distributed/pipelining/test_transformer.py:83]: @drisspg @albanD any other idea about this? Is the change to add pu1 variant proper here?

**Review by drisspg — APPROVED** [review]

**Comment by PlatformLC** [comment]
@fffrog @drisspg In addition, I've modified the change to add hw-classification for test cases, PTAL again.

**Review by jishuangfeng — APPROVED** [review]
LGTM. 
On the allow_xpu question — it's a historical API quirk: XPU and MPS were added to core after the fact and require explicit opt-in (allow_xpu/allow_mps), while CPU/CUDA/HPU/PrivateUse1 are included by default in get_device_type_test_bases. The PrivateUse1 variant here is correct — it was already in the default bases, the old only_for whitelist was just filtering it out.

**Review by fffrog — APPROVED** [review]
SGTM, thank you.
